include/alp/structures.hpp

```diff
@@ -666,6 +674,14 @@ namespace alp {
 
 		};
 
+		template<>
+		struct isInstantiable< UpperTrapezoidal, UpperTrapezoidal > {
+			template< typename ImfR, typename ImfC >
+			static bool check( const ImfR &imf_r, const ImfC &imf_c ) {
+				return imf_r.map( 0 ) <= imf_c.map( 0 );
```

Comment:
I think we should make sure that `imf_r.map( 0 ) <= imf_c.map( n - 1 )` ?

Reply:
Is `n` here `imf_c.n`? What would this test? The current condition is checking that the diagonal is taken above the original diagonal.

Reply:
If not mistaken it would make sure you are not selecting a complete zero region.